### PR TITLE
Implement check prevention

### PR DESCRIPTION
**Pull request by khalifaumar308**

### CODE REVIEW

File: engine.py [hunk 1]
```diff
@@ -196,7 +201,7 @@ def make_move(self, move):
 					master.destroy()
 					break
 
-	def undo_move(self, look_ahead_mode = False):
+	def undo_move(self, move, n, look_ahead_mode = False, undo=True):
```
Comment (drak00) [inline engine.py]: what is n?

**Review by drak00 — CHANGES_REQUESTED** [review]
check prevention only works for light king. Please fix the logic for dark king

File: engine.py (hunk 1 above)
Comment (khalifaumar308) [reply engine.py]: Is a parameter that allows undo_move to print only if u key is pressed

Comment (drak00) [reply engine.py]: Making your parameter names explicit helps with code readability. You could name it "allow_undo" for example

**Pull request merged**